### PR TITLE
Logging impl

### PR DESCRIPTION
**Pull request by dymayday**
This is the first implementation of the **asynchronous** `Logger`.

Related to this issue #29 

From the top of my head, here is a list of the things this logger will implement:
- [x] Simple logging using macros
- [x] Use config file
- [x] Log to file

### CODE REVIEW

File: logging/examples/features.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+pub fn main() {
+    let _guard = Logger::new();
+
+    trace!("logging a trace message");
+    debug!("debug values"; "x" => 1, "y" => -1);
+    info!("some interesting info"; "where" => "right here");
+    warn!("be cautious!"; "why" => "you never know...");
+    error!("wrong {}", "foobar"; "type" => "unknown");
+    crit!("abandoning test");
+}
```
Comment (struktured) [inline logging/examples/features.rs]: @dymayday I find the the call instantiating the logger here a bit suspect. I feel either the method `new()` is misleading and instead should be named to imply a singleton pattern or that we register a logger reference with the macro system in a more explicit manner. Your thoughts?

File: logging/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+/// This struct is used as a wrapper around the `slog` crate for our customized logger.
+/// This has to be instantiate at the top of the user program in order to keep
+/// the logging thread alive.
+pub struct Logger(slog_scope::GlobalLoggerGuard);
+
+impl Logger {
+    pub fn new() -> Self {
+        Logger::term_logger()
+    }
+
```
Comment (struktured) [inline logging/src/lib.rs]: If this is a singleton it probably shouldn't be named `new()`. Is it setup / teardown style? Like an `init()` method? Or more like a wrapper function around a singleton? Eg. `get()` or `instance()` or `try...()`?

Comment (dymayday) [reply logging/src/lib.rs]: I wanted to stick to idiomatic Rust with a `new` constructor, but indeed, calling a `init()` is probably more accurate, thanks for your input on this.

File: logging/src/lib.rs [hunk 3]
```diff
@@ -1 +1,351 @@
...
+        let tmp_file = NamedTempFile::new().expect("Fail to create temporary file.");
+        let tmp_file_path = tmp_file.into_temp_path();
+        let tmp_file_path = tmp_file_path
+            .to_str()
+            .expect("Fail to convert 'PathBuf' to 'String'.");
+
+        let _guard = Lib3hLogger::init_file_logger(&tmp_file_path);
+        info!("{}", dumped_log_string);
+
+        // // This is the messed up part...
```
Comment (zippy) [inline logging/src/lib.rs]: These comments should be fixed or removed.

**Comment by neonphog** [comment]
Closing this, feel free to re-open if needed.

**Pull request closed without merging**